src/Android/Main.cpp

```diff
@@ -289,9 +289,10 @@ JNIEXPORT void JNICALL
 Java_org_xcsoar_NativeView_pauseNative(JNIEnv *env, jobject obj)
 {
   if (event_queue == nullptr || CommonInterface::main_window == nullptr)
-    /* pause before we have initialized the event subsystem does not
-       work - let's bail out, nothing is lost anyway */
-    exit(0);
+  {
+    /* event subsystem is not initialized, there is nothing to pause */
+    return;
```

Comment:
Did you verify that this is safe to do?

Reply:
Again consulting the android [reference](https://developer.android.com/guide/components/activities/activity-lifecycle) I don't see why this would be unsafe.
The call to `onPause` seems to be a chance to stop or reduce activities that are not needed while the activity is not in the foreground. There is no direct mention of anything that would be unsafe to do while in this state. I think the activity would simply continue in the background.

Reply:
The problem isn't Android - the problem is the big number of global variables which aren't initalized properly in XCSoar's very old code base. Doing `main()` again in a reused process may break lots of things, that's why we just `exit()` on Android to prevent reusing the process.
Of course, it would be a good idea to get rid of all global variables, and over the past 11 years (since I've joined), I have removed most of them, but not all.
I was asking because I thought maybe you were doing this in a corner case where recycling the process doesn't matter.

Reply:
I see.
If I understand correctly, the first time `onResume` is called, `initSDL` initializes the NativeView. The case that this return is hit happens when the NativeView has been created but hasn't finished yet. So as soon as `nativeView` contains the object the second `onResume` does not call `initSDL` again.
So I would think that no global variables would be changed, as no "second start" took place.

Reply:
I took another look today and one thing this causes is that the MainWindow is started while the Activity is in the paused state. When returning from the permissions dialog `CommonInterface::main_window->Resume()` is called.
This checks whether it has really been paused and just returns otherwise, so I don't see anything bad happening there either.

Reply:
Agree, I think you're right.